tests/Go/TestUtils/JoinPointsExtractor.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2011, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\TestUtils;
+
+use PhpParser\Node;
+use PhpParser\NodeTraverser;
+use PhpParser\Node\Stmt;
+use PhpParser\NodeVisitorAbstract;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+use PhpParser\PrettyPrinter\Standard as Printer;
+
+/**
+ * Extracts join points definitions from woven class proxy by parsing and traversing class AST.
+ */
+final class JoinPointsExtractor extends NodeVisitorAbstract
```

Comment:
Wow, looks very complicated for me.

Why not just to use `ReflectionClass::getStaticPropertyValue()` for asking the value of static property in the class? Of course, it will be already replaced with joinpoints, but it will be much easier than using AST for that task.

Alternatively, if you still want to parse source file, then give this job to `goaop/parser-reflection`, it's already on board. Just ask the same `getStaticPropertyValue()` to get this value for free :)